### ISSUE TITLE
ESLint fails to detect all arguments of a function as defined but never used

### ISSUE THREAD

**Issue by plumlee** [eslint/eslint#6316]
**What version of ESLint are you using?**
2.9.x, also tested with 2.11.x using npm link. Using eslint as part of xo@0.15.1.

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLint I believe - the issue I’ve seen is using xo@0.15.1.

**Please show your full configuration:**
Using config via xo:

    "space": true,
    "envs": [
        "browser"
    ]

**What did you do? Please include the actual source code causing the issue.**

This is part of a Mocha+Chai+Expect test spec where I see the issue:

```
    it('should convert the arguments of a function to an array', function () {
      var testFn = function (a, b, c) {
        return utils.toArray(arguments);
      };
      expect(testFn(1, 2, 3)).to.deep.equal([1, 2, 3]);
    });
```

Running xo and thus eslint as above outputs:
`error  'c' is defined but never used  no-unused-vars`

Remove the third argument:
```
    it('should convert the arguments of a function to an array', function () {
      var testFn = function (a, b) {
        return utils.toArray(arguments);
      };
      expect(testFn(1, 2)).to.deep.equal([1, 2]);
    });
```

Run as above outputs:
`error  ‘b' is defined but never used  no-unused-vars`

And then the same if you take it all the way down to just `a`.

**What did you expect to happen?**

I expect eslint to detect all function arguments are defined but never used. ESLint appears to be detecting only the last item in the function arguments, regardless of the number of arguments. This makes me suspect that the issue is either a RegExp failing to match properly or something iterating through function arguments but failing to properly increment and capture errors.

**What actually happened? Please include the actual, raw output from ESLint.**
See above.

I’m happy to start digging into the code and do a PR if someone can point me in the right direction as to where to start.

**Comment by ilyavolodin** [comment]
This is correct behavior. While none of the arguments are used, in a lot of cases, you have to specify a few unused arguments to get to the one you want to use. For example, in a lot of cases, callback signatures, but you only want to use 3-rd one. Unless you use `arguments` property, you have no way to only declare one parameter and expect it to get 3-rd value passed into the function.
This is configurable and is controlled by `args` option of the rule (http://eslint.org/docs/rules/no-unused-vars#args). If you want all parameters to be checked, you can configure the rule in the following matter:
```js
"no-unused-vars": ["error", { "args": "all" }]
```
This will force the rule to check all arguments.

Closing, as this is behaving as expected. Feel free to reopen if you disagree.
